src/main/java/com/yash/ExpenseClaims/controllers/FileUploadController.java

```diff
@@ -0,0 +1,49 @@
+package com.yash.ExpenseClaims.controllers;
+
+import com.yash.ExpenseClaims.dto.ExpenseDto;
+import com.yash.ExpenseClaims.repositories.UploadFilesToDB;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.ModelAndView;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+@RestController
+public class FileUploadController {
+
+    @Autowired
+    private UploadFilesToDB uploadFilesToDB;
+
+    @PostMapping("/upload")
+    public ModelAndView handleFileUpload(
+            @RequestParam("username") String username,
+            @RequestParam("amount") Double amount,
+            @RequestParam("reason") String reason,
+            @RequestParam("file") MultipartFile file) {
+
+        uploadFilesToDB.upload(username, amount, reason, file);
+        HashMap<String, String> variables = new HashMap<>();
```

Comment:
use [ModelMap](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/ui/ModelMap.html) so that we can do chaining like fluent API